### PR TITLE
Upgrade to Grafana 10x and MermaidJS 10x

### PR DESCRIPTION
**Pull request by xaner4**
This PR builds on #244 to upgrade Grafana plugin to support Grafana 10x.
And also upgraded MermaidJS version 10x

### CODE REVIEW

**Review by rekha86sundararajan — APPROVED** [review]

**Comment by xaner4** [comment]
Hi @jdbranham.
Can this PR be merged into master and if not can you please advice if there's something I can change or do differently to get it merged?

File: src/config/types.ts [hunk 1]
```diff
@@ -45,14 +45,9 @@ export interface MetricCharacterReplacement {
...
-  displayName: string;
-}
-
 export interface CompositeMetric {
   name: string;
-  members: CompositeMember[];
+  members: string[];
   valueName: ValueType;
   showLowestValue: boolean;
 }
```
Comment (jdbranham) [inline src/config/types.ts:53]: Do composite metrics work for you?

**Comment by jdbranham** [comment]
Nice work!

One question about the Composite Metrics. It doesn't seem to work for me. 
Are they working for you?

File: src/config/types.ts (hunk 1 above)
Comment (jdbranham) [reply src/config/types.ts:53]: Maybe a question for @rekha86sundararajan since it was part of the original PR

Comment (jdbranham) [reply src/config/types.ts:53]: It was actually part of the PR I already merged. oops.
 
@xaner4  If you'll rebase this on master, we can go ahead and merge it up!

Comment (xaner4) [reply src/config/types.ts:53]: Should we have a look at the CompositeMetric before doing any thing or will you just merge it in as is?

I don't think think there's any thing to rebase into master as my PR already builds upon the latest PR you merged.
But i'm no git expert at any means so maybe i'm wrong?

Comment (xaner4) [reply src/config/types.ts:53]: Any thing left to do to get this merged @jdbranham?

**Comment by jdbranham** [comment]
Hi @xaner4 - 
I pulled the branch locally and it works great, although I get an error on the first load.  
I think it has something to do with the deprecated `stylesFactory` but not sure yet.  
Can you take a look?  Or I can dig a bit deeper later this week 👍

File: src/DiagramController.tsx [hunk 2]
```diff
@@ -35,7 +34,7 @@ interface DiagramPanelControllerState {
 const getDiagramWithLegendStyles = stylesFactory(({ options }: DiagramPanelControllerProps) => ({
   wrapper: css`
```
Comment (jdbranham) [inline src/DiagramController.tsx:35]: I think the error is happening when this is called.  🤔

**Pull request merged**